scripts/k8s_deploy_rapids_dask.sh

```diff
@@ -1,127 +1,231 @@
+  pushd "${RAPIDS_TMP_BUILD_DIR}"
 
   # Build the docker image
-  docker build -t dask-rapids
+  docker build --network=host -t ${DASK_IMAGE} .
 
   popd
-  rm -rf tmp-rapids-build
-
-
-  # TODO: Push the docker  image
-
+  rm -rf "${RAPIDS_TMP_BUILD_DIR}"
+  if [ "${DOCKER_PUSH}" != "" ]; then
+    docker tag ${DASK_IMAGE} ${DOCKER_REGISTRY}/${DASK_IMAGE}
+    docker push ${DOCKER_REGISTRY}/${DASK_IMAGE}
+  fi
 }
 
+
 function tear_down() {
-  echo "Tearing down existing resources"
   # Delete existing resources
-  helm list ${NAMESPACE} | grep ${NAMESPACE} 2> /dev/null 1> /dev/null
-  if [ "${?}" == "0" ]; then
-    read -r -p "Helm resources already exist, would you  like to delete them? (yes/no)" response
+  if [ "$(helm list ${RAPIDS_HELM_NAME})" != "" ]; then
+    read -r -p "Helm resources already exist, would you  like to delete them (this will include associated non-helm k8s service accounts as well)? (yes/no/skip)" response
     case "$response" in
       [yY][eE][sS]|[yY])
-        helm delete --purge ${NAMESPACE}
-	;;
+        helm delete --purge ${RAPIDS_HELM_NAME}
+        kubectl delete -n ${RAPIDS_NAMESPACE} -f config/k8s/rapids-dask-sa.yml || true # Best effort to delete any existing service accounts in the ns
+        sleep 2
+        ;;
+      skip)
+        echo "Continuing with no cleanup"
+        ;;
       *)
         echo "Quitting install"
-	exit
-	;;
+        exit
+        ;;
     esac
-  fi	  
-  kubectl get ns rapids 2> /dev/null 1> /dev/null
-  if [ "${?}" == "0" ]; then
-    read -r -p "Kubernetes resources already exist, would you  like to delete them? (yes/no)" response
+  fi
+
+  if [ "$(kubectl get ns ${RAPIDS_NAMESPACE})" != "" -a "${RAPIDS_SKIP_KUBERNETES}" != "true"  ]; then
+    read -r -p "Kubernetes resources already exist, would you  like to delete them? (yes/no/skip)" response
     case "$response" in
       [yY][eE][sS]|[yY])
-        kubectl delete ns ${NAMESPACE}
-	;;
+        kubectl delete ns ${RAPIDS_NAMESPACE}
+        sleep 2
+        ;;
+      skip)
+        echo "Continuing with no cleanup"
+        ;;
       *)
         echo "Quitting install"
-	exit
-	;;
+        exit
+        ;;
     esac
-  fi	  
-  sleep 5
+  fi
 }
 
+
 function stand_up() {
   echo "installing dask via helm"
+
   # Create the Dask resources
-  helm install -n ${NAMESPACE} --namespace ${NAMESPACE} --values config/helm/rapids-dask.yml stable/dask
-  kubectl create -f config/k8s/rapids-dask-sa.yml
+  helm install -n ${RAPIDS_HELM_NAME} --namespace ${RAPIDS_NAMESPACE} --values config/helm/rapids-dask.yml ${DASK_CHART_NAME}
+  kubectl create -n ${RAPIDS_NAMESPACE}  -f config/k8s/rapids-dask-sa.yml
+
   # kubectl create -f  config/k8s/rapids-dask-autoscale.yml # XXX: Optional install
-  sleep 5 # TODO: This should poll
+  while [ `kubectl -n ${RAPIDS_NAMESPACE} get pods | grep Running | wc -l` != ${pod_count} ]; do
```

Comment:
Nit: instead of doing this loop, could we use `kubectl wait` as in https://github.com/NVIDIA/deepops/blob/ef5d44c01c14f326d7bc716c3c9b62c24c7b5ece/scripts/k8s_deploy_ingress.sh#L31?

Reply:
I'll push this in now. I wasn't able to get the kubectl wait command to work earlier, but that looks more robust.